ea_server/api/utils.py

```diff
@@ -0,0 +1,59 @@
+import functools
+import math
+def transform(result, data):
+    result_as_dict = dict()
+    for order_index,p in enumerate(result):
+        driver_index = math.floor(p)
+        driver_id = data["drivers"][driver_index]["id"]
+        order_in_route = p-driver_index
+        item=(data["orders"][order_index]["id"],p)
+        
+        if data["drivers"][driver_index]["id"] in result_as_dict:
+            addToOrderedList(result_as_dict[driver_id],item)
+        else:
+            orders_list = list()
+            orders_list.append(item)
+            result_as_dict[driver_id]= orders_list
+
+    for driver in result_as_dict:
+        result_as_dict[driver]= refactor_orders_list(result_as_dict[driver])
+
+    return result_as_dict
+
+def refactor_orders_list(orders_list: list):
+    res = list()
```

Comment:
I suggest you to read about 'List Comprehension' in python.

It's really cool and saves a lot of time. 🙃